app/controllers/administrate/application_controller.rb

```diff
@@ -76,7 +76,12 @@ def destroy
       else
         flash[:error] = requested_resource.errors.full_messages.join("<br/>")
       end
-      redirect_to after_resource_destroyed_path(requested_resource)
+      after_action = after_resource_destroyed_path(requested_resource)[:action]
+      if valid_action?(after_action, resource_class)
+        redirect_to after_resource_destroyed_path(requested_resource)
+      else
+        redirect_to params[:origin_url]
+      end
```

Comment:
This code assumes that `after_resource_destroyed_path` always returns a hash with an `:action` key. This will probably not be the case when someone overrides it. (And it was created specifically to be overridden).

We need some other way to tell if the route exists. Perhaps we can do something with `url_for`? (No idea from the top of my head).

Alternative idea: only provide `origin_url` if the show page doesn't exist. This means checking the existence of the route when rendering the link, instead of when following the link. Then we always follow `origin_url` if present.